compilers/amd.js

```diff
@@ -360,34 +263,17 @@ exports.remap = function(source, map, fileName) {
 
 
 // converts anonymous AMDs into named AMD for the module
-exports.compile = function(load, opts, loader) {
-  var normalize = opts.normalize;
-  var options = { sourceRoot: true, script: true };
-  if (opts.sourceMaps)
-    options.sourceMaps = 'memory';
-  if (opts.lowResSourceMaps)
-    options.lowResolutionSourceMap = true;
-
-  if (load.metadata.sourceMap)
-    options.inputSourceMap = load.metadata.sourceMap;
-
-  var compiler = new traceur.Compiler(options);
-
-  var tree = load.metadata.parseTree || compiler.parse(load.source, load.path);
-  var transformer = new AMDDefineRegisterTransformer(!opts.anonymous && load.name, load, load.metadata.anonDefine, load.metadata.anonDefineIndex, normalize ? load.depMap : {});
-  tree = transformer.transformAny(tree);
-
-  var output = compiler.write(tree, load.path);
-
-  // AMD define extraction via parsing stops on var define redefinitions
-  // so this creates a natural boundary to allow future folds of this same code through rebundling
-  return Promise.resolve({
-    source: '(function() {\nvar define = ' + opts.systemGlobal + '.amdDefine;\n' + output + '\n})();',
-    sourceMap: compiler.getSourceMap(),
-    sourceMapOffset: 2
-  });
+exports.compile = function (load, opts, loader) {
+  opts.moduleId = !opts.anonymous && load.name; 
+  return compiler.compile(load, opts, [require('babel-plugin-transform-amd-system-wrapper').default, {
+    map: function (dep) {
+      return opts.normalize ? load.depMap[dep] : dep;
+    },
+    systemGlobal: opts.systemGlobal,
+	deps: load.deps
```

Comment:
Initially i didn't used this dependencies. But some unit tests still failed, and i had no idea what's wrong. Then i looked into the original code and found this code snippet: https://github.com/systemjs/builder/blob/master/compilers/amd.js#L222

It sems there are additional deps only available in the loader metadata.